tools/tests/test_nd_freeze

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+# Temporary scratch directory to save generated files.
+TMP_DIR="tmp_$((1 + RANDOM % 10000))"
+
+# Run common commands for each test at start of tests
+#
+# Parameters
+# ----------
+# distro : string
+#   Valid values are "debain" or "ubuntu"
+# release : string
+#   Code name for distro release. (e.g. jessie)
+# date : string
+#   Date at which the snapshot of the OS is wanted
+# keep_sources : string (Optional)
+#   Pass the string "KEEP_SOURCES" to run nd_freeze with the --keep-original-apt-sources switch
+#
+function test_setup {
+    distro=$1
+    release=$2
+    date=$3
+    keep_sources=${4:-no}
+
+    if [ "$keep_sources" = "no" ]; then
```

Comment:
since no "standard" test runner is used, it would be nice to add here some printout about what is going on, since otherwise it just sits quietly at "Test basic operation ..." for a while